src/models/Team.ts

```diff
@@ -27,20 +32,53 @@ export default class Team {
     }
 
     get currentSize(): number{
-        return this._playerList.length;
+        return this._playerList.length + this._substitutionPlayerList.length;
+    }
+
+    get isFull(): boolean {
+        return this.currentSize == this.targetSize;
     }
 
     addPlayer(player: Player): boolean {
-        if (!this._playerList.includes(player) && this.currentSize < this.targetSize){
+        if (!this.isFull && !this.isPlayerInTeam(player)){
```

Comment:
The whole function can be simplified by:
return (!this.isFull && !this.isPlayerInTeam(player)) ? (this._playerList.push(player), true) : false;

Reply:
Done.